session.go

```diff
@@ -699,9 +703,23 @@ func (q *Query) defaultsFromSession() {
 	q.serialCons = s.cfg.SerialConsistency
 	q.defaultTimestamp = s.cfg.DefaultTimestamp
 	q.idempotent = s.cfg.DefaultIdempotence
+	q.metrics = make(map[string]*queryMetrics)
 	s.mu.RUnlock()
 }
 
+func (q *Query) getHostMetrics(host *HostInfo) *queryMetrics {
+	q.session.mu.Lock()
```

Comment:
What is this lock protecting?

Reply:
@annismckenzie You're right, I am working on speculative queries executions, and indeed I use wait groups to keep track of the parallel query executions. The lock here is protecting from the parallel access of the attempts number in the query from different goroutines ....but mind you that I've built the speculative stuff a week before I started this work, and since I've discovered a few holes in my understanding of how retryPolicies and retries in general work. So much could change  and I might remove the locks after all.